osbs/build/build_requestv2.py

```diff
@@ -305,6 +310,26 @@ def set_data_from_reactor_config(self, reactor_config_data):
         self.set_registry_organization(reactor_config_data)
         self.set_required_secrets(reactor_config_data)
 
+    def set_build_env_vars(self, reactor_config_data):
+        """
+        Propagates build_env_vars from config map to build environment
+        """
+        env = self.template['spec']['strategy']['customStrategy']['env']
+        existing_vars = set(var['name'] for var in env)
+
+        build_env_vars = reactor_config_data.get('build_env_vars', [])
+        for var in build_env_vars:
+            if var['name'] not in existing_vars:
+                # Could be just `env.append(var)`, but here we are using the reactor config
+                # without validating it (that happens later, in atomic-reactor). Do it this
+                # way to fail early if format does not match schema.
+                env.append({'name': var['name'], 'value': var['value']})
+                existing_vars.add(var['name'])
+                logger.info('Set environment variable from reactor config: %s', var['name'])
+            else:
```

Comment:
I'd prefer fail, this warning will nobody notice and this shouldn't happen

Reply:
Good point, changed